quic/s2n-quic/src/tests.rs

```diff
@@ -691,3 +699,109 @@ fn increasing_pto_count_under_loss() {
         pto_count
     );
 }
+
+// TODO: https://github.com/aws/s2n-quic/issues/1726
+//
+// The rustls tls provider is used on windows and has different
+// build options than s2n-tls. We should build the rustls provider with
+// mTLS enabled and remove the `cfg(target_os("windows"))`.
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn mtls() {
+    use crate::provider::{event::events::HandshakeStatus, tls::default as s2n_tls};
+
+    let max_handshake_duration = Duration::from_secs(5);
+    let limits = provider::limits::Limits::default()
+        .with_max_handshake_duration(max_handshake_duration)
+        .unwrap();
+
+    // Ensure connection is successful under different network conditions
+    for delay_time in (500..=10_000).step_by(1000) {
+        for drop_percent in (0..=100).step_by(10) {
+            let delay_time = Duration::from_micros(delay_time);
+
+            let model = Model::default();
+            model.set_delay(delay_time);
+            let server_subscriber = HandshakeStatusRecorder::new();
+            let server_status = server_subscriber.events();
+            let client_subscriber = HandshakeStatusRecorder::new();
+            let client_status = client_subscriber.events();
+
+            test(model.clone(), |handle| {
+                spawn(async move {
+                    // allow for 1 RTT worth of data before impairing the connection.
+                    // Both the Client and Server have received initial packets at
+                    // this point.
+                    delay(delay_time * 2).await;
+
+                    // simulate network impairment
+                    let drop_rate = drop_percent as f64 / 100.0;
+                    model.set_drop_rate(drop_rate);
+
+                    // restore the network after approximately half the max handshake
+                    // timeout to allow the handshake to succeed
+                    delay(max_handshake_duration / 2).await;
+                    model.set_drop_rate(0.0);
```

Comment:
So I choose the max_handshake / 2 after some experimentation. Its hard to pinpoint the sleep duration since there seems to be some interplay between the network delay, sleep and max handshake duration.

However, since we are testing all the various combinations I think it sets a good baseline for us to work from. Thoughts?

Reply:
Spoke offline. The delay and loss wasnt providing much value and I decided to remove it. Instead I added a success and failure case for mTLS testing.